### PR TITLE
Fix DependencyContext splitting on semi-colon

### PR DESCRIPTION
**Pull request by eerhardt**
https://github.com/dotnet/runtime/commit/5e67657e20665c32c2bd5c4ac1c8b1af78c9677e introduced a bug in DependencyContextPaths where the static array is not initialized before it is being used in the Create static method.

This fix removes the static array since it is only used once.

### CODE REVIEW

**Review by ViktorHofer — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextPaths.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@ internal sealed class DependencyContextPaths
         private const string DepsFilesProperty = "APP_CONTEXT_DEPS_FILES";
         private const string FxDepsFileProperty = "FX_DEPS_FILE";
 
+        private static readonly char[] s_semicolon = new[] { ';' };
```
Comment (stephentoub) [inline src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextPaths.cs:15]: Do we even want to cache this? If DependencyContextPaths.Create is only called from GetCurrent, and GetCurrent is only called from that initializer, then that Split call happens only once, and caching this is actually a deoptimization.

Comment (eerhardt) [reply src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextPaths.cs]: Yeah, I was just fixing that. 😄. See updated commit.

File: src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextPaths.cs [hunk 2]
```diff
@@ -42,7 +40,13 @@ private static DependencyContextPaths GetCurrent()
 
         internal static DependencyContextPaths Create(string? depsFiles, string? sharedRuntime)
         {
-            string[]? files = depsFiles?.Split(s_semicolon, StringSplitOptions.RemoveEmptyEntries);
+#if NETCOREAPP
+            const char separator = ';';
+#else
+            // This method is only executed at startup. No need to cache the char[].
```
Comment (stephentoub) [inline src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextPaths.cs:46]:
```suggestion
            // This method is only executed once at startup. No need to cache the char[].
```

File: src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextPaths.cs [hunk 3]
```diff
@@ -42,7 +40,13 @@ private static DependencyContextPaths GetCurrent()
 
         internal static DependencyContextPaths Create(string? depsFiles, string? sharedRuntime)
```
Comment (stephentoub) [inline src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextPaths.cs:41]: Can this be private?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextPaths.cs (hunk 3 above)
Comment (eerhardt) [reply src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextPaths.cs:41]: No, unit tests use it through IVT. They could be refactored to use reflection instead, but I'm trying to get this through to unblock code flow - see https://github.com/dotnet/installer/pull/16621#issuecomment-1587705679.

Comment (stephentoub) [reply src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextPaths.cs:41]: Oh IVT, how you mock me.

**Comment by eerhardt** [comment]
Build time outs are tracked by https://github.com/dotnet/runtime/issues/76454.

System.Collections.Tests failing on WASI appear to be:

* https://github.com/dotnet/runtime/issues/81911
* https://github.com/dotnet/runtime/issues/82043

File: src/libraries/Microsoft.Extensions.DependencyModel/tests/DependencyContextTests.cs [hunk 4]
```diff
@@ -276,6 +276,19 @@ public void MergeMergesRuntimeGraph()
                 Subject.Fallbacks.Should().BeEquivalentTo("win7-x64", "win7-x86");
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "GetEntryAssembly() returns null")]
+        public void DefaultWorksCorrectly()
```
[the remark this answers is not in the document]
Comment (eerhardt) [reply src/libraries/Microsoft.Extensions.DependencyModel/tests/DependencyContextTests.cs:281]: Fixing in https://github.com/dotnet/runtime/pull/87666.

**Pull request merged**